Improve worker queues overview, other minor improvements (#1411)
astronomer/docs: astro/ci-cd.md

import TabItem from '@theme/TabItem';
import {siteVariables} from '@site/src/versions';

Continuous Integration and Continuous Delivery (CI/CD) is an industry term that refers to programmatic workflows that automate key parts of the software development lifecycle, including code changes, builds, and testing. CI/CD enables teams to develop faster, more securely, and more reliably.

On Astro, you can use Deployment API keys to automate deploying code changes to a Deployment. Astronomer recommends setting up a CI/CD workflow for all production environments.

> This guide has CI/CD templates that can be used with popular CI/CD management tools, including GitHub Actions and Circle CI.

Reviewer: ```suggestion
You can use the Astronomer CI/CD templates with popular CI/CD management tools, including GitHub Actions and Circle CI.
```

## Benefits

There are many benefits to configuring a CI/CD workflow that automates pushing code changes to Astro. Specifically, you can:

> - Deploy new and updated DAGs in a way that streamlines the development process among team members.

Reviewer: ```suggestion
- Deploy new and updated DAGs in a way that streamlines your development process.
```

Reviewer: How?

## CI/CD templates

> This section provides basic templates for configuring individual CI pipelines using popular CI/CD tools. Each template can be implemented as-is to produce a simple CI/CD pipeline, but Astronomer recommends reconfiguring the templates to work with your own directory structures, workflows, and best practices. More templates are coming soon.

Reviewer: ```suggestion
Templates allow you to quickly configure individual CI pipelines using popular CI/CD tools. Each template can be implemented as-is to produce a simple CI/CD pipeline. Astronomer recommends reconfiguring the templates to work with your own directory structures, workflows, and best practices.
```

> At a high level, the CI/CD templates:

Reviewer: ```suggestion
The CI/CD templates:
```
